src/lib/libcore.js

```diff
@@ -2296,6 +2296,74 @@ addToLibrary({
 
   $noExitRuntime: "{{{ makeModuleReceiveExpr('noExitRuntime', !EXIT_RUNTIME) }}}",
 
+  // The following addOn<X> functions are for adding runtime callbacks at
+  // various executions points. Each addOn<X> function has a corresponding
+  // compiled time version named addAt<X> that will instead inline during
+  // compilation (see parseTools.mjs).
+  // Note: if there are both runtime and compile time code, the runtime
+  // callbacks will be invoked before the compile time code.
+
+  // See ATPRERUNS in parseTools.mjs for more information.
+  $__ATPRERUN__: [],
+  $__ATPRERUN____deps: ['$callRuntimeCallbacks'],
+  $__ATPRERUN____postset: () => {
+    ATPRERUNS.unshift('callRuntimeCallbacks(__ATPRERUN__);');
+  },
+  $addOnPreRun__deps: ['$__ATPRERUN__'],
+  $addOnPreRun: (cb) => {
+    __ATPRERUN__.unshift(cb);
+  },
+  // See ATINITS in parseTools.mjs for more information.
+  $__ATINIT__: [],
```

Comment:
Do these arrays need to be visible to user code?   If not, maybe we can mark all these arrays as `__internal: true`?

Also, perhaps we could give them better names without the double underscores?   Now sure what the best thing might be?   `runtimeCallbacksOnInit` ?   Or something shorter:  `onInits`?

Reply:
They don't need to be visible. I'll add and rename.